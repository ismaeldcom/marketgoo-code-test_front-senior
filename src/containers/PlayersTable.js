import React, { useEffect } from 'react'
import { useSelector, useDispatch } from 'react-redux'
import { Table, TableRow, TableCell, Button } from '@marketgoo/ola'
import TablePlaceholder from 'components/TablePlaceholder'
import { removePlayer, getPlayers } from 'store/players'

const PlayersTable = () => {
    const dispatch = useDispatch()
    const players = useSelector(state => state.players)

    useEffect(() => {
        dispatch(getPlayers())
    }, [])

    return (
        <>
            <Table
                responsive
                caption={
                    players && players.length === 0 && 'No players created'
                }
            >
                <thead>
                    <TableRow>
                        <TableCell header>Player</TableCell>
                        <TableCell header>Team</TableCell>
                        <TableCell header variant='numeric'>
                            Score
                        </TableCell>
                        <TableCell header variant='right'>
                            Actions
                        </TableCell>
                    </TableRow>
                </thead>
                <tbody>
                    {!players ? (
                        <TablePlaceholder />
                    ) : (
                        players.map(player => (
                            <TableRow key={player.id || player.name}>
                                <TableCell>{player.name}</TableCell>
                                <TableCell>{player.team}</TableCell>
                                <TableCell variant='numeric'>
                                    {player.score}
                                </TableCell>
                                <TableCell variant='right'>
                                    <Button
                                        variant='destructive'
                                        onClick={() =>
                                            dispatch(removePlayer(player))
                                        }
                                    >
                                        Remove
                                    </Button>
                                </TableCell>
                            </TableRow>
                        ))
                    )}
                </tbody>
            </Table>
        </>
    )
}

export default PlayersTable
