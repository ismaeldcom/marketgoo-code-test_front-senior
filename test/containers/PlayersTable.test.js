import React from 'react'
import renderer from 'react-test-renderer'
import { useSelector } from 'react-redux'
import PlayersTable from 'containers/PlayersTable'

jest.mock('react-redux', () => ({
    useDispatch: () => jest.fn(),
    useSelector: jest.fn()
}))

describe('Snapshots', () => {
    describe('PlayersTable', () => {
        it('renders placeholder', () => {
            useSelector.mockReturnValue(null)
            const tree = renderer.create(<PlayersTable />).toJSON()
            expect(tree).toMatchSnapshot()
        })

        it('renders player', () => {
            useSelector.mockReturnValue([
                {
                    id: 1,
                    name: 'Peter',
                    team: 'Pandas',
                    score: 1
                }
            ])
            const tree = renderer.create(<PlayersTable />).toJSON()
            expect(tree).toMatchSnapshot()
        })
    })
})
