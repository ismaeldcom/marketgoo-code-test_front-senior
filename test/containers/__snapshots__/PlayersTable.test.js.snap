// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshots PlayersTable renders placeholder 1`] = `
<div
  className="ola_table-container is-responsive"
>
  <table
    className="ola_table"
  >
    <thead>
      <tr
        className="ola_tableRow"
      >
        <th
          className="ola_tableCell is-left"
        >
          Player
        </th>
        <th
          className="ola_tableCell is-left"
        >
          Team
        </th>
        <th
          className="ola_tableCell is-numeric"
        >
          Score
        </th>
        <th
          className="ola_tableCell is-right"
        >
          Actions
        </th>
      </tr>
    </thead>
    <tbody>
      <tr
        className="ola_tableRow"
      >
        <td
          className="ola_tableCell is-left"
        >
          <div
            className="placeholder"
          />
        </td>
        <td
          className="ola_tableCell is-left"
        >
          <div
            className="placeholder"
          />
        </td>
        <td
          className="ola_tableCell is-left"
        >
          <div
            className="placeholder"
          />
        </td>
        <td
          className="ola_tableCell is-left"
        >
          <div
            className="placeholder"
          />
        </td>
      </tr>
      <tr
        className="ola_tableRow"
      >
        <td
          className="ola_tableCell is-left"
        >
          <div
            className="placeholder"
          />
        </td>
        <td
          className="ola_tableCell is-left"
        >
          <div
            className="placeholder"
          />
        </td>
        <td
          className="ola_tableCell is-left"
        >
          <div
            className="placeholder"
          />
        </td>
        <td
          className="ola_tableCell is-left"
        >
          <div
            className="placeholder"
          />
        </td>
      </tr>
      <tr
        className="ola_tableRow"
      >
        <td
          className="ola_tableCell is-left"
        >
          <div
            className="placeholder"
          />
        </td>
        <td
          className="ola_tableCell is-left"
        >
          <div
            className="placeholder"
          />
        </td>
        <td
          className="ola_tableCell is-left"
        >
          <div
            className="placeholder"
          />
        </td>
        <td
          className="ola_tableCell is-left"
        >
          <div
            className="placeholder"
          />
        </td>
      </tr>
      <tr
        className="ola_tableRow"
      >
        <td
          className="ola_tableCell is-left"
        >
          <div
            className="placeholder"
          />
        </td>
        <td
          className="ola_tableCell is-left"
        >
          <div
            className="placeholder"
          />
        </td>
        <td
          className="ola_tableCell is-left"
        >
          <div
            className="placeholder"
          />
        </td>
        <td
          className="ola_tableCell is-left"
        >
          <div
            className="placeholder"
          />
        </td>
      </tr>
      <tr
        className="ola_tableRow"
      >
        <td
          className="ola_tableCell is-left"
        >
          <div
            className="placeholder"
          />
        </td>
        <td
          className="ola_tableCell is-left"
        >
          <div
            className="placeholder"
          />
        </td>
        <td
          className="ola_tableCell is-left"
        >
          <div
            className="placeholder"
          />
        </td>
        <td
          className="ola_tableCell is-left"
        >
          <div
            className="placeholder"
          />
        </td>
      </tr>
    </tbody>
  </table>
</div>
`;

exports[`Snapshots PlayersTable renders player 1`] = `
<div
  className="ola_table-container is-responsive"
>
  <table
    className="ola_table"
  >
    <thead>
      <tr
        className="ola_tableRow"
      >
        <th
          className="ola_tableCell is-left"
        >
          Player
        </th>
        <th
          className="ola_tableCell is-left"
        >
          Team
        </th>
        <th
          className="ola_tableCell is-numeric"
        >
          Score
        </th>
        <th
          className="ola_tableCell is-right"
        >
          Actions
        </th>
      </tr>
    </thead>
    <tbody>
      <tr
        className="ola_tableRow"
      >
        <td
          className="ola_tableCell is-left"
        >
          Peter
        </td>
        <td
          className="ola_tableCell is-left"
        >
          Pandas
        </td>
        <td
          className="ola_tableCell is-numeric"
        >
          1
        </td>
        <td
          className="ola_tableCell is-right"
        >
          <button
            className="ola_button is-destructive"
            disabled={false}
            onClick={[Function]}
          >
            <span
              className="ola_button-text"
            >
              Remove
            </span>
          </button>
        </td>
      </tr>
    </tbody>
  </table>
</div>
`;
